controller/array_action/array_mediator_svc.py

```diff
@@ -294,6 +297,51 @@ def get_host_by_host_identifiers(self, initiators):
             logger.debug("can not found host by using initiators: {0} ".format(initiators))
             raise controller_errors.HostNotFoundError(initiators)
 
+    def _get_detailed_hosts_list(self):
+        logger.debug("Getting detailed hosts list on array {0}".format(self.endpoint))
+        hosts_list = self.client.svcinfo.lshost()
+        if not hosts_list:
+            return []
+
+        # for each host get its detailed info from array by sending batch of commands
+        detailed_hosts_list_cmd = self._get_detailed_hosts_list_cmd(hosts_list)
+        logger.debug("Sending getting detailed hosts list commands batch")
+        detailed_hosts_list_output, detailed_hosts_list_errors = self.client.send_raw_command(detailed_hosts_list_cmd)
+        detailed_hosts_list_as_string = bytes_to_string(detailed_hosts_list_output)
+        detailed_hosts_list_error_msg = self._get_formatted_hosts_list_error_msg(detailed_hosts_list_errors)
+        if not detailed_hosts_list_errors:
```

Comment:
why `not`? perhaps we need a UT for this...

Reply:
then let me rephrase: why does `if not errors` mean that "there are errors"?

Reply:
We don't have UT to check logs

Reply:
it shouldn't be hard, but could you address my question? it's still not clear to me

Reply:
If not <empty list>

Reply:
ok. I guess it's good enough

Reply:
fixed

Reply:
thanks! but I still think the fix should include a Unit Test (to avoid this confusion in the future).
if you think it's too much for this ticket, then at least open a ticket and add it here

Reply:
Resonsider about log assert here CSI-1492